git_machete/git_operations.py

```diff
@@ -796,7 +798,12 @@ def is_equivalent_tree_reachable(
             return True
 
         if (equivalent_to_commit_hash, reachable_from_commit_hash) in self.__is_equivalent_tree_reachable_cached:
-            return self.__is_equivalent_tree_reachable_cached[equivalent_to_commit_hash, reachable_from_commit_hash]
+            prev_mode, prev_result = self.__is_equivalent_tree_reachable_cached[equivalent_to_commit_hash, reachable_from_commit_hash]
+
+            # Only return cached result if we're using the same mode or if we already checked with the
```

Comment:
I don't think this extra caching check makes sense. IIUC within the given git-machete invocation (i.e. within the given Python process lifetime), there's only ever a single set of CLI options, and hence a single squash detection mode. I'll probably revert that.

Reply:
And within tests, a new `GitContext` is created in each test method anyway.

Reply:
I ended up splitting the caches into one for equivalent-tree-reachable and one for equivalent-patch-reachable